administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
+          {
+            $dst_frames[$key] = imagecreatetruecolor($destWidth, $destHeight);
+          }
         }
         else
         {
-          if($imginfo[2] == 'PNG')
-          {
-            $src_img = imagecreatefrompng($src_file);
-          }
-          else
-          {
-            $src_img = imagecreatefromgif($src_file);
-          }
+          $dst_frames[0] = imagecreatetruecolor($destWidth, $destHeight);
+          $src_frames = JoomFile::imageCreateFrom_GD($src_file, $dst_frames, $imginfo);
         }
-
-        if(!$src_img)
+        if (in_array(false, $src_frames))
         {
+          $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_LIBARY_NOT_ABLE_RESIZING');
           return false;
         }
-
         if($angle > 0)
+        // rotate image, if needed
         {
-          $src_img = imagerotate($src_img, $angle, 0);
+          foreach ($src_frames as $key => $frame)
+          {
+            $src_frames[$key]['image'] = imagerotate($src_frames[$key]['image'], $angle, 0);
+          }
         }
-
-        $dst_img = imagecreatetruecolor($destWidth, $destHeight);
-
-        if($config->jg_fastgd2thumbcreation == 0)
+        foreach ($src_frames as $key => $frame)
         {
-          if(!is_null($offsetx) && !is_null($offsety))
+          if($config->jg_fastgd2thumbcreation == 0)
+          // use normal GD2 for resizing
           {
-            imagecopyresampled( $dst_img, $src_img, 0, 0, $offsetx, $offsety,
-                                $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
+            if(!is_null($offsetx) && !is_null($offsety))
+            {
+              imagecopyresampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, $offsetx, $offsety,
+                                  $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
+            }
+            else
+            {
+              imagecopyresampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, 0, 0, $destWidth,
+                                  (int)$destHeight, $srcWidth, $srcHeight);
+            }
           }
           else
+          // use fast GD2 for resizing
           {
-            imagecopyresampled( $dst_img, $src_img, 0, 0, 0, 0, $destWidth,
-                                (int)$destHeight, $srcWidth, $srcHeight);
+            if(!is_null($offsetx) && !is_null($offsety))
+            {
+              $dst_frames[$key] = JoomFile::fastImageCopyResampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, $offsetx, $offsety,
+                                                                    $destWidth, (int)$destHeight, $srcWidth, $srcHeight, 3,$imginfo);
+            }
+            else
+            {
+              $dst_frames[$key] = JoomFile::fastImageCopyResampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, 0, 0, $destWidth,
+                                                                    (int)$destHeight, $srcWidth, $srcHeight, 3,$imginfo);
+            }
```

Comment:
The resizing is exact the same as before. Except it is performed for each frame of the src_frames array.